tests/api_jwe.c

```diff
@@ -98,5 +107,13 @@ main(int argc, char *argv[])
     assert(dec(jwe, set1));
     assert(dec(jwe, set2));
 
+    char* long_str_300k = get_string(300000, 'a');
+    json_decref(jwe);
+    assert((jwe = json_object()));
+    assert(jose_jwe_enc(NULL, jwe, NULL, jwke, long_str_300k, 300000));
+    assert(!dec(jwe, jwke));
```

Comment:
This should fail only for compressed payloads, uncompressed payloads should be totally unaffected.
So you need a specific test that does use deflate for the payload.

Reply:
Hello @simo5 . Thanks for detecting this. You are completely right. I will re code it.